src/rewrite.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { RewriteRule, RewriteRules } from "./rewrite";
+
+describe("RewriteRule", () => {
+  describe("without capture groups", () => {
+    it("applies the expected changes", () => {
+      const rule = new RewriteRule(/-[a-z0-9]+-?/gi, "");
```

Comment:
**nit:** Could we add a comment explaining the regex to the expressions in these tests? Will help identify the intention and to verify the output is expected